lib/shoryuken/manager.rb

```diff
@@ -19,8 +19,11 @@ def initialize(fetcher, polling_strategy)
       @polling_strategy = polling_strategy
 
       @heartbeat = Concurrent::TimerTask.new(run_now: true,
-                                             execution_interval: HEARTBEAT_INTERVAL,
-                                             timeout_interval: 60) { dispatch }
+                                             execution_interval: HEARTBEAT_INTERVAL) { @pool.post { dispatch } if @dispatching.false? }
```

Comment:
@waynerobinson thanks for reviewing it 🍻 

I've just changed it to use a separate and single thread pool.

I'm now running some performance tests, let's see how it goes 🙏